pages/index.js

```diff
@@ -84,19 +84,44 @@ export default function Home() {
         return e;
       });
       setPairs(_pairs);
+
+      //transform to tokens
+      const tokensMap = _pairs.reduce((acc, curr) => {
+        const token0 = curr.token0.contractAddress;
+        const token1 = curr.token1.contractAddress;
+        const reserve0Usd = +curr.token0Reserve * curr.token0.usdPrice;
+        const reserve1Usd = +curr.token1Reserve * curr.token1.usdPrice;
+        const volume0 = +curr.token0Volume * curr.token0.usdPrice;
+        const volume1 = +curr.token1Volume * curr.token1.usdPrice;
+        
+        return {
+          ...acc,
+          [token0]: {
+            ...curr.token0,
+            tvl: (acc[token0]?.tvl ?? 0) + reserve0Usd,
+            volume: (acc[token0]?.volume ?? 0) + volume0,
+          },
+          [token1]: {
+            ...curr.token1,
+            tvl: (acc[token1]?.tvl ?? 0) + reserve1Usd,
+            volume: (acc[token1]?.volume ?? 0) + volume1,
+          },
+        };
+      }, {});
+      setTokens(Object.values(tokensMap).filter(e => e.tvl));
     };
     getPairs();
   }, []);
 
-  useEffect(() => {
-    const getTokens = async () => {
-      const tokensApiResult = await axios(`${API_BASE_URL}tokens`);
-      const _tokens = Object.values(tokensApiResult.data);
-      setTokens(_tokens);
-    }
+  // useEffect(() => {
+  //   const getTokens = async () => {
+  //     const tokensApiResult = await axios(`${API_BASE_URL}tokens`);
+  //     const _tokens = Object.values(tokensApiResult.data);
+  //     setTokens(_tokens);
+  //   };
 
-    getTokens();
-  }, []);
+  //   getTokens();
+  // }, []);
```

Comment:
Do we want to leave old code as a comment, does it need to be brought back in the future or why is it here?

Reply:
I removed it